### PR TITLE
bitnami/external-dns Add support for OCI IAM instance principle and workload identity

### LINKED ISSUE #24459: [bitnami/external-dns] Add support for OCI IAM Instance Principall and OCI IAM Workload Identity
### Name and Version

bitnami/external-dns 6.37.0

### What is the problem this feature will solve?

The Oracle provider for external DNS supports alternative authentication methods.

https://github.com/kubernetes-sigs/external-dns/blob/master/docs/tutorials/oracle.md#oci-iam-instance-principal
https://github.com/kubernetes-sigs/external-dns/blob/master/docs/tutorials/oracle.md#oci-iam-workload-identity

These don't require creating an OCI user with a long-lived key and secret.

It seems this chart does not support these methods as it always tries to create a secret and mount. https://github.com/bitnami/charts/blob/main/bitnami/external-dns/templates/dep-ds.yaml#L836
https://github.com/bitnami/charts/blob/main/bitnami/external-dns/templates/dep-ds.yaml#L769

it was also reported in this https://github.com/bitnami/charts/pull/9334

### What is the feature you are proposing to solve the problem?

Update https://github.com/bitnami/charts/blob/main/bitnami/external-dns/templates/dep-ds.yaml#L769 to include else statements to skip if alternative authentication methods are used.

### What alternatives have you considered?

_No response_

### PR DESCRIPTION
**Pull request by cloudoutloud**
### Description of the change

Raised to address https://github.com/bitnami/charts/issues/24459

Adds support when using the OCI provider to use different authentication methods.

If using [oci-iam-instance-principal](https://github.com/kubernetes-sigs/external-dns/blob/master/docs/tutorials/oracle.md#oci-iam-instance-principal) It will enable via flag and not add oci-config-file volume and mount.

If using [oci-iam-workload-identity](https://github.com/kubernetes-sigs/external-dns/blob/master/docs/tutorials/oracle.md#oci-iam-workload-identity) it will still mount volume with secret but the content is stripped down.

### Benefits

We can use alternative authentication methods when using the OCI provider instead of using the default method of creating a user with long-lived API keys.

### Applicable issues

- fixes #24459

### Checklist

- [x] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/). This is *not necessary* when the changes only affect README.md files.
- [x] Variables are documented in the values.yaml and added to the `README.md` using [readme-generator-for-helm](https://github.com/bitnami/readme-generator-for-helm)
- [x] Title of the pull request follows this pattern [bitnami/<name_of_the_chart>] Descriptive title
- [x] All commits signed off and in agreement of [Developer Certificate of Origin (DCO)](https://github.com/bitnami/charts/blob/main/CONTRIBUTING.md#sign-your-work)

### CODE REVIEW

**Comment by carrodher** [comment]
Thanks for your contribution, could you please check some issues related to the metadata present in values.yaml in charge of generating the README table? See https://github.com/bitnami/charts/actions/runs/8448383659/job/23140364581?pr=24708
```
######
The following errors must be fixed before proceeding
######

ERROR: Missing metadata for key: oci.useInstancePrincipal
ERROR: Missing metadata for key: oci.useWorkloadIdentity
ERROR: Metadata provided for non existing key: oci.enableInstancePrincipal
ERROR: Metadata provided for non existing key: oci.enableWorkloadIdentity
```

**Comment by cloudoutloud** [comment]
@carrodher metadata issue has been fixed

File: bitnami/external-dns/templates/dep-ds.yaml [hunk 1]
```diff
@@ -282,6 +282,12 @@ spec:
             - --infoblox-max-results={{ .Values.infoblox.maxResults }}
               {{- end }}
             {{- end }}
+            {{- if eq .Values.provider "oci" }}
```
Comment (jotamartos) [inline bitnami/external-dns/templates/dep-ds.yaml:285]: Use "if and" here 

```
{{- if and (eq .Values.provider "oci") .Values.useInstancePrincipal }}
```

**Review by jotamartos — COMMENTED** [review]
Your changes LGTM. Please take a look at my comment

**Comment by cloudoutloud** [comment]
@jotamartos in regards to testing these changes with the image, what is the best way?

**Comment by jotamartos** [comment]
>@jotamartos in regards to testing these changes with the image, what is the best way?

Please follow the documentation of the application and confirm everything works as expected. 

https://github.com/kubernetes-sigs/external-dns/blob/master/docs/tutorials/oracle.md

As described in our documentation, we perform a battery of tests to every solution we publish. In this case, this is a specific configuration of the application and it needs to be tested manually. 

https://github.com/bitnami/charts/blob/main/TESTING.md

Thanks

**Comment by cloudoutloud** [comment]
@jotamartos I have tested these changes manually with the application and a DNS hosted zone in Oracle works as expected.

**Review by jotamartos — APPROVED** [review]

**Pull request merged**